zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/KerberosInterpreter.java

```diff
@@ -48,23 +48,38 @@ public KerberosInterpreter(Properties property) {
   @ZeppelinApi
   protected abstract boolean runKerberosLogin();
 
-  public String getKerberosRefreshInterval() {
+  @ZeppelinApi
+  protected abstract boolean isKerboseEnabled();
+
+  public void open() {
+    if (isKerboseEnabled()) {
+      startKerberosLoginThread();
+    }
+  }
+
+  public void close() {
+    if (isKerboseEnabled()) {
+      shutdownExecutorService();
+    }
+  }
+
+  private String getKerberosRefreshInterval() {
     if (System.getenv("KERBEROS_REFRESH_INTERVAL") == null) {
```

Comment:
Sure, have added isKerboseEnabled in javadoc. 
For `KERBEROS_REFRESH_INTERVAL` I think all interpreters can share a common its value defined in `zeppelin-env.sh` (https://github.com/apache/zeppelin/blob/master/conf/zeppelin-env.sh.template#L58-L59) instead of having it at interpreter level, and increasing the complexity (number of lines) of interpreter config.